template/CHART_NAME/templates/deployment.yaml

```diff
@@ -0,0 +1,110 @@
+      {{- end }}
+      {{- if .Values.%%MAIN_OBJECT_BLOCK%%.priorityClassName }}
+      priorityClassName: {{ .Values.%%MAIN_OBJECT_BLOCK%%.priorityClassName | quote }}
+      {{- end }}
+      {{- if .Values.%%MAIN_OBJECT_BLOCK%%.podSecurityContext.enabled }}
+      securityContext:
+        fsGroup: {{ .Values.%%MAIN_OBJECT_BLOCK%%.podSecurityContext.fsGroup }}
+      {{- end }}
+      {{- if .Values.%%MAIN_OBJECT_BLOCK%%.initContainers }}
+      initContainers: {{- include "common.tplvalues.render" (dict "value" .Values.%%MAIN_OBJECT_BLOCK%%.initContainers "context" $) | nindent 8 }}
+      {{- end }}
+      containers:
+        - name: %%CONTAINER_NAME%%
+          image: {{ template "%%TEMPLATE_NAME%%.image" . }}
+          imagePullPolicy: {{ .Values.%%MAIN_OBJECT_BLOCK%%.image.pullPolicy }}
+          {{- if .Values.%%MAIN_OBJECT_BLOCK%%.lifecycleHooks }}
+          lifecycle: {{- include "common.tplvalues.render" (dict "value" .Values.%%MAIN_OBJECT_BLOCK%%.lifecycleHooks "context" $) | nindent 12 }}
+          {{- end }}
+          {{- if .Values.%%MAIN_OBJECT_BLOCK%%.containerSecurityContext.enabled }}
+          securityContext:
+            runAsUser: {{ .Values.%%MAIN_OBJECT_BLOCK%%.containerSecurityContext.runAsUser }}
+            runAsNonRoot: {{ .Values.%%MAIN_OBJECT_BLOCK%%.containerSecurityContext.runAsNonRoot }}
+          {{- end }}
+          command:
+          {{- if .Values.%%MAIN_OBJECT_BLOCK%%.command }}
+          {{- include "common.tplvalues.render" (dict "value" .Values.%%MAIN_OBJECT_BLOCK%%.command "context" $) | nindent 12 }}
+          {{- else }}
+            %%CONTAINER_COMMAND%%
+          {{- end }}
+          {{- if .Values.%%MAIN_OBJECT_BLOCK%%.args }}
+          args: {{- include "common.tplvalues.render" (dict "value" .Values.%%MAIN_OBJECT_BLOCK%%.args "context" $) | nindent 12 }}
+          {{- end }}
+          env:
+            - name: foo
+              value: bar
+          envFrom:
+            {{- if .Values.%%MAIN_OBJECT_BLOCK%%.extraEnvVarsCM }}
+            - configMapRef:
+                name: {{ include "common.tplvalues.render" (dict "value" .Values.%%MAIN_OBJECT_BLOCK%%.extraEnvVarsCM "context" $) }}
+            {{- end }}
+            {{- if .Values.%%MAIN_OBJECT_BLOCK%%.extraEnvVarsSecret }}
+            - secretRef:
+                name: {{ include "common.tplvalues.render" (dict "value" .Values.%%MAIN_OBJECT_BLOCK%%.extraEnvVarsSecret "context" $) }}
+            {{- end }}
+          {{- if .Values.%%MAIN_OBJECT_BLOCK%%.resources }}
+          resources: {{- toYaml .Values.%%MAIN_OBJECT_BLOCK%%.resources | nindent 12 }}
+          {{- end }}
+          {{- if .Values.%%MAIN_OBJECT_BLOCK%%.customLivenessProbe }}
+          livenessProbe: {{- include "common.tplvalues.render" (dict "value" .Values.%%MAIN_OBJECT_BLOCK%%.customLivenessProbe "context" $) | nindent 12 }}
+          {{- end }}
```

Comment:
This is OK but I think that the most common approach is checking for a default liveness/readinessProbe and else use the custom one:

e.g. (Kong)

```yaml
{{- if .Values.kong.livenessProbe.enabled }}
livenessProbe:
  exec:
    command:
      - /bin/bash
      - -ec
      - /health/kong-container-health.sh
  initialDelaySeconds: {{ .Values.kong.livenessProbe.initialDelaySeconds }}
  periodSeconds: {{ .Values.kong.livenessProbe.periodSeconds }}
  timeoutSeconds: {{ .Values.kong.livenessProbe.timeoutSeconds }}
  failureThreshold: {{ .Values.kong.livenessProbe.failureThreshold }}
  successThreshold: {{ .Values.kong.livenessProbe.successThreshold }}
{{- else if .Values.kong.customLivenessProbe }}
livenessProbe: {{- include "common.tplvalues.render" (dict "value" .Values.kong.customLivenessProbe "context" $) | nindent 12 }}
{{- end }}
```
Do you agree?